addons/sale_project/models/project.py

```diff
@@ -119,7 +121,9 @@ def _compute_display_sales_stat_buttons(self):
 
     def action_view_sols(self):
         self.ensure_one()
-        all_sale_order_lines = self._fetch_sale_order_items({'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]})
+        all_sale_order_lines = self._fetch_sale_order_items(
+            {'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]}).search(
+            [('company_id', '=', self.env.company.id)])
```

Comment:
There is no issue in that action. 

However, we got a traceback in the method below (`action_view_sos`) because of we fetch the order_id field in that line:  
`all_sale_orders = self._fetch_sale_order_items({'project.task': [('state', 'in', self.env['project.task'].OPEN_STATES)]}).order_id`

Reply:
I think both actions need to be modified, no? Anything that returns IDs of SOLs must filter based on company, but I've missed the `action_view_sos` action. I will add the filter.

Reply:
Answered below.